From 2b5acba1c14b19cc48f8e90310d5d52e6bead931 Mon Sep 17 00:00:00 2001
From: Ed Falk <efalk@cyngn.com>
Date: Thu, 1 Oct 2015 17:59:09 -0700
Subject: [PATCH 44/51] FM: Fix settings UI bugs

Setting any user-defined parameter caused the band to switch
from user-defined to Indonesia. Fixed.
Also removed some bogus localization while I was there.

Issue-Id: CYNGNOS-778
Change-Id: I256f04eab39b820b33676bf598322d480d6dfa44
---
 fmapp2/src/com/caf/fmradio/Settings.java | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/fmapp2/src/com/caf/fmradio/Settings.java b/fmapp2/src/com/caf/fmradio/Settings.java
index 0897ad4..55410fc 100644
--- a/fmapp2/src/com/caf/fmradio/Settings.java
+++ b/fmapp2/src/com/caf/fmradio/Settings.java
@@ -144,6 +144,7 @@ public class Settings extends PreferenceActivity implements
            mUserBandMinPref.getEditText().setInputType(InputType.TYPE_CLASS_NUMBER |
                                                         InputType.TYPE_NUMBER_FLAG_DECIMAL);
            mUserBandMinPref.setDialogTitle(R.string.usr_def_band_min);
+           mUserBandMinPref.getEditText().setSelectAllOnFocus(true);
 
            mUserBandMaxPref = new EditTextPreference(this);
            mUserBandMaxPref.setKey(USER_DEFINED_BAND_MAX_KEY);
@@ -151,6 +152,7 @@ public class Settings extends PreferenceActivity implements
            mUserBandMaxPref.setDialogTitle(R.string.usr_def_band_max);
            mUserBandMaxPref.getEditText().setInputType(InputType.TYPE_CLASS_NUMBER |
                                                         InputType.TYPE_NUMBER_FLAG_DECIMAL);
+           mUserBandMaxPref.getEditText().setSelectAllOnFocus(true);
 
            setBandSummary(index);
 
@@ -344,14 +346,10 @@ public class Settings extends PreferenceActivity implements
                   && (noOfChannels > 0) && (band_width >= 100)) {
                   FmSharedPreferences.setLowerLimit((int)freq);
                   sendSettingsChangedIntent(FM_BAND_CHANGED);
-                  setBandSummary(summaryBandItems.length - 1);
+                  setBandSummary(FmSharedPreferences.REGIONAL_BAND_USER_DEFINED);
                   clearStationList();
                }else {
-                  if ((Locale.getDefault().toString().equals("zh_HK")))
-                      Toast.makeText(this,"請輸入有效的頻道範圍76.0-108.0",
-                                                            Toast.LENGTH_SHORT).show();
-                  else
-                      Toast.makeText(this, getString(R.string.user_defind_band_msg),
+                  Toast.makeText(this, getString(R.string.user_defind_band_msg),
                                                             Toast.LENGTH_SHORT).show();
                }
            }else if(key.equals(USER_DEFINED_BAND_MAX_KEY)) {
@@ -372,14 +370,10 @@ public class Settings extends PreferenceActivity implements
                   && (noOfChannels > 0) && (band_width >= 100)) {
                   FmSharedPreferences.setUpperLimit((int)freq);
                   sendSettingsChangedIntent(FM_BAND_CHANGED);
-                  setBandSummary(summaryBandItems.length - 1);
+                  setBandSummary(FmSharedPreferences.REGIONAL_BAND_USER_DEFINED);
                   clearStationList();
                }else {
-                  if ((Locale.getDefault().toString().equals("zh_HK")))
-                      Toast.makeText(this,"請輸入有效的頻道範圍76.0-108.0",
-                                                            Toast.LENGTH_SHORT).show();
-                  else
-                      Toast.makeText(this, getString(R.string.user_defind_band_msg),
+                  Toast.makeText(this, getString(R.string.user_defind_band_msg),
                                                             Toast.LENGTH_SHORT).show();
                }
           }else {
-- 
2.17.1

